libraries/eosiolib/dispatcher.hpp

```diff
@@ -124,13 +124,9 @@ namespace eosio {
 extern "C" { \
    void apply( uint64_t receiver, uint64_t code, uint64_t action ) { \
       eosio::name self(receiver); \
-      if( action == eosio::name("onerror").value) { \
-         /* onerror is only valid if it is for the "eosio" code account and authorized by "eosio"'s "active permission */ \
-         eosio_assert(code == eosio::name("eosio").value, "onerror action's are only valid from the \"eosio\" system account"); \
-      } \
-      if( code == self.value || action == eosio::name("onerror").value ) { \
+      if( code == self.value ) { \
```

Comment:
For consistency we might as well remove the line above and change this line to `if( code == receiver ) {`.

Then in `EOSIO_DISPATCH_HELPER` the call to `execute_action` would be:
`eosio::execute_action( eosio::name(receiver), eosio::name(code), &OP::elem );`